test/test_alert_types.cpp

```diff
@@ -0,0 +1,59 @@
+/*
+
+Copyright (c) 2017, Arvid Norberg, Alden Torres
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions
+are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in
+      the documentation and/or other materials provided with the distribution.
+    * Neither the name of the author nor the names of its
+      contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+POSSIBILITY OF SUCH DAMAGE.
+
+*/
+
+#include "libtorrent/alert_manager.hpp"
+#include "libtorrent/alert_types.hpp"
+#include "test.hpp"
+#include "setup_transfer.hpp"
+
+using namespace libtorrent;
+
+TORRENT_TEST(dht_get_peers_reply_alert)
+{
+	alert_manager mgr(1, dht_get_peers_reply_alert::static_category);
+
+	TEST_EQUAL(mgr.should_post<dht_get_peers_reply_alert>(), true);
+
+	sha1_hash const ih = rand_hash();
+	tcp::endpoint const ep1 = rand_tcp_ep();
+	tcp::endpoint const ep2 = rand_tcp_ep();
+	std::vector<tcp::endpoint> const v = {ep1, ep2};
+
+	mgr.emplace_alert<dht_get_peers_reply_alert>(ih, v);
+
+	auto const* a = alert_cast<dht_get_peers_reply_alert>(mgr.wait_for_alert(seconds(0)));
+	TEST_CHECK(a != nullptr);
+
+	TEST_EQUAL(a->info_hash, ih);
+	TEST_EQUAL(a->num_peers(), 2);
+	TEST_CHECK(a->peers() == v);
```

Comment:
in which case this assertion would not necessarily be right, but sorting both vectors before comparing should make it pass